functorch/_src/aot_autograd.py

```diff
@@ -567,7 +1105,106 @@ def backward(ctx, *flat_args):
+                    # Because the original_inpt might be a view of some larger tensor,
+                    # and updated_inpt is always densely packed.
+                    if original_inpt.size() != updated_inpt.size() \
+                            or original_inpt.stride() != updated_inpt.stride() \
+                            or original_inpt.storage_offset() != updated_inpt.storage_offset():
+                        # Functionalization can't easily tell us if an input had BOTH its metadata actual data mutated.
+                        # So we check if metadata needs to be mutated here manually.
+                        original_inpt.as_strided_(updated_inpt.size(), updated_inpt.stride(), updated_inpt.storage_offset())
+                    original_inpt.copy_(updated_inpt)
+        else:
+            fw_outs = outs
+
+        # Step 4: Manually regenerate any outputs that are aliased to inputs, instead of
+        # compiling them.
+        if CompiledFunction.num_aliased_outputs > 0:
+            assert CompiledFunction.num_aliased_outputs + len(fw_outs) == len(CompiledFunction.fw_metadata.aliased_output_info)
+            fw_outs_including_aliases = []
+            curr_fw_out_idx = 0
+            for maybe_aliased_out_metadata in CompiledFunction.fw_metadata.aliased_output_info:
+                if maybe_aliased_out_metadata is None:
+                    fw_outs_including_aliases.append(fw_outs[curr_fw_out_idx])
+                    curr_fw_out_idx += 1
+                else:
+                    input_alias_idx, out_tensor_meta = maybe_aliased_out_metadata
+                    input_alias = args[input_alias_idx]
+                    # Note: here, we manually regenerate the output, using an as_strided() call,
+                    # OR if the aliased output came from a custom autograd.function, we replay it.
+                    # The as_strided() in the normal case is good for perf (this is hot-path code,
+                    # and we're consolidating potential chains of views into a single view op).
+                    # But we might need to figure out view replaying for e.g. XLA.
+                    # TODO: handle the custom autograd function case here.
+                    # We need a way to check whether a tensor came from a custom autograd fn from python,
+                    # AND a way to replay that custom view fn.
+                    if out_tensor_meta is None:
+                        # This handles the specific case where the user returns an output that *was* an input. Don't create a view.
+                        fw_outs_including_aliases.append(input_alias)
+                    else:
+                        # TODO: when the aot autograd cache lands, we'll need to change this
+                        # to evaluate the sympy expressions given the inputs?
+                        size_ = [guard_int(x) for x in out_tensor_meta.size()]
```

Comment:
is it worth passing the file/line into guard_int for traceability?

Reply:
that would be a good point - fortunately, I'm killing all of this in the refactor :)